src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1218,3 +1221,20 @@
     (when (fs/exists? cert)
       (fs/delete cert)
       (log/debug "Deleted certificate for" subject))))
+
+(schema/defn ^:always-validate get-custom-oid-mappings :- (schema/maybe OIDMappings)
+  "Given a path to a custom OID mappings file, return a map of all oids to
+  shortnames"
+  [custom-oid-mapping-file :- (schema/maybe schema/Str)]
+  (when (fs/file? custom-oid-mapping-file)
```

Comment:
I agree. I actually based this on the parsing of the `csr_attributes` file, which we're already doing in clojure code, so I think we should do this there as well. I'll push up a commit that adds logging in both cases.